### PR TITLE
PP-950: PP-929: PBS Pro (windows) services are not able to start due to missing libical dll's

### PR DESCRIPTION
**Pull request by bremanandjk**
#### Issue-ID

* **[PP-950](https://pbspro.atlassian.net/browse/PP-950)**
* **[PP-929](https://pbspro.atlassian.net/browse/PP-929)**

#### Problem description
* Windows PBS Pro services are not coming up properly.

#### Cause / Analysis
* Dynamic way of building "libical" third party library missed to add new dll's into the msi package.
This caused the PBS PRO services not to start properly due to missing libraries. 

#### Solution description
* Fix would be packaging new dll's and setting environment PATH to link it correctly.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: win_configure/msi/wix/prep.bat [hunk 1]
```diff
@@ -88,6 +88,7 @@ if %ERRORLEVEL% == 0 (
 		"%binaries_path%\tcltk\bin\*.dll"
 	) do (
 	xcopy /s /d "%%~a" "%EXECDIR%\bin" > NUL
+	if NOT %ERRORLEVEL% == 0 goto theend
 	)
 ) else (
 echo Fail to create %EXECDIR%\bin directory.
```
Comment (sid2364) [inline win_configure/msi/wix/prep.bat]: Could you change this to "Failed to create .."? 
Also throughout the rest of the file.

**Comment by bremanandjk** [comment]
Thanks @sid2364 
I have pushed another commit with required changes. Plz have a look !

**Comment by bremanandjk** [comment]
Thanks @sid2364 for your approval, 
@PBSPro/pbspro-maintainers , please take this PR further. Thanks

**Comment by bremanandjk** [comment]
There is issue with opensuse travis CI, til yesterday, it was working fine. 

Now, builds are bailing out with below error.

> +docker exec -i testdev zypper -n ref
> 
> Timeout exceeded when accessing 'http://download.opensuse.org/repositories/devel:/tools/openSUSE_Leap_42.3/media.1/media'.
> 
> Retrying in 30 seconds...
> 
> Download (curl) error for 'http://download.opensuse.org/repositories/devel:/tools/openSUSE_Leap_42.3/repodata/repomd.xml':
> 
> Error code: HTTP response: 500
> 
> Error message: The requested URL returned error: 500 Internal Server Error
> 
> Abort, retry, ignore? [a/r/i/...? shows all options] (a): a
> 
> ABORT request: Aborting requested by user
> 
> The command "./.travis/${OS_TYPE//:/_}.sh" failed and exited with 4 during .
> 
> Your build has been stopped.

File: win_configure/msi/wix/prep.bat [hunk 2]
```diff
@@ -88,9 +88,10 @@ if %ERRORLEVEL% == 0 (
 		"%binaries_path%\tcltk\bin\*.dll"
 	) do (
 	xcopy /s /d "%%~a" "%EXECDIR%\bin" > NUL
+	if NOT %ERRORLEVEL% == 0 goto theend
 	)
 ) else (
-echo Fail to create %EXECDIR%\bin directory.
+echo Failed to create %EXECDIR%\bin directory.
 goto theend
```
Comment (subhasisb) [inline win_configure/msi/wix/prep.bat]: Ident this lines properly

Comment (bremanandjk) [reply win_configure/msi/wix/prep.bat]: Thanks @subhasisb  for your review comments, 

I have pushed new commit with changes, please have a look.

**Comment by hirenvadalia** [comment]
@bremanandjk While I was working on fixing windows build failure due to whatever changes we did to support AppVeyor, I tried to fix installer issues but due to time shortage I couldn't complete that...
And luckily I have kept copy of whatever changes I did, which might be useful to you and might save some of your time 😄...
So if you want, you can have a look at [msi/wix](https://github.com/hirenvadalia/pbspro/tree/winbuild_back/win_configure/msi/wix) and [win_postinstall.py](https://github.com/hirenvadalia/pbspro/blob/winbuild_back/src/cmds/scripts/win_postinstall.py)

PS: Please feel free to copy-paste and ask me any question if you have any doubt in changes...

**Comment by bremanandjk** [comment]
@hirenvadalia , Thanks for sharing code changes, It really helps !
Shall i create a new ticket like "Improvements in MSI installer" and merge-in your changes. Since this PP-950 changes are must-fix to make MSI installer to function properly. 

Already two approval are in-place, its ready to merge. What you say?

**Comment by subhasisb** [comment]
@bremanandjk please squash your commits for the merge

**Comment by subhasisb** [comment]
@bremanandjk @hirenvadalia i suggest we file those enhancements as a separate ticket and address as part of that.

**Comment by hirenvadalia** [comment]
@bremanandjk I agree we should address in separate request.

**Comment by bremanandjk** [comment]
@subhasisb I have squashed my commits, please take further. Thanks

**Comment by bremanandjk** [comment]
@subhasisb @hirenvadalia 
Thanks for agreeing, I have created new ticket [PP-962](https://pbspro.atlassian.net/browse/PP-962) for enhancements.

**Comment by subhasisb** [comment]
@bremanandjk please rebase for merge

**Comment by bremanandjk** [comment]
@subhasisb I have rebased again, please take further, Thanks

**Pull request merged**